### PR TITLE
Feature/yearn actions

### PR DESCRIPTION
**Pull request by majkic99**

### CODE REVIEW

File: contracts/actions/yearn/YearnWithdraw.sol [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            _inputData.amount = amountPulled;
+
+            address underlyingToken = vault.token();
+
+            uint256 underlyingTokenBalanceBefore = underlyingToken.getBalance(address(this));
+            vault.withdraw(_inputData.amount);
+            uint256 underlyingTokenBalanceAfter = underlyingToken.getBalance(address(this));
+            tokenAmountReceived = sub(underlyingTokenBalanceAfter, underlyingTokenBalanceBefore);
+
+            if (address(this) != _inputData.to){
```
Comment (yippee-ki-yay) [inline contracts/actions/yearn/YearnWithdraw.sol:74]: Handle if _inputData.to is 0x0

File: contracts/interfaces/yearn/yVault.sol [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.7.6;
+
+interface YVault {
```
Comment (yippee-ki-yay) [inline contracts/interfaces/yearn/yVault.sol:5]: Interfaces should start with I

File: contracts/interfaces/yearn/YearnRegistry.sol [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.7.6;
+
+abstract contract YearnRegistry {
```
Comment (yippee-ki-yay) [inline contracts/interfaces/yearn/YearnRegistry.sol:5]: IYearnRegistry

File: contracts/actions/yearn/YearnSupply.sol [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        logger.Log(
+                address(this),
+                msg.sender,
+                "YearnSupply",
+                abi.encode(_inputData, yTokenAmount)
+            );
+    }
+
+    /// @dev this function sends whole proxy yToken balance to an address
+    function _yearnDirectSupply(Params memory _inputData) internal {
```
Comment (yippee-ki-yay) [inline contracts/actions/yearn/YearnSupply.sol:88]: Avoid code duplication, reuse this in `_yearnSupply `

File: contracts/actions/yearn/YearnSupply.sol [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @dev this function sends only the amount of yTokens received from supplying to an account
+    /// @return yTokenAmount amount of yTokens we received for supplying
+    function _yearnSupply(Params memory _inputData) internal returns (uint256 yTokenAmount) {
+        YVault vault = YVault(yearnRegistry.latestVault(_inputData.token));
+
+        uint amountPulled = _inputData.token.pullTokensIfNeeded(_inputData.from, _inputData.amount);
```
Comment (yippee-ki-yay) [inline contracts/actions/yearn/YearnSupply.sol:66]: If amount uint.max pull whole balance _from?

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Things look pretty good, try and remove code duplication and a few small fixes and we are good to go.

File: contracts/actions/yearn/YearnSupply.sol (hunk 5 above)
Comment (majkic99) [reply contracts/actions/yearn/YearnSupply.sol:64]: yes, tokenUtils handles uint_max and puts whole balance as return value

File: contracts/actions/yearn/YearnWithdraw.sol [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+/// @title Supplies tokens to Yearn vault
+contract YearnWithdraw is ActionBase, DSMath {
+    using TokenUtils for address;
+
+    /// @param token - address of yToken to withdraw (same as yVault address)
+    /// @param amount - amount of yToken to withdraw
+    /// @param from - address from which to pull tokens from
+    /// @param to - address where received underlying tokens will be sent to
+    struct Params {
+        address token;
```
Comment (yippee-ki-yay) [inline contracts/actions/yearn/YearnWithdraw.sol:20]: Maybe yToken and yAmount is better so we know it's not the underlying token

**Review by yippee-ki-yay — APPROVED** [review]
Lgtm

**Pull request merged**